src/client.rs

```diff
@@ -45,11 +53,54 @@ impl Client {
             _ => Channel::balance_list(endpoints.into_iter()),
         };
 
-        let kv = KvClient::new(channel.clone(), None);
-        let watch = WatchClient::new(channel.clone(), None);
-        let lease = LeaseClient::new(channel, None);
+        let mut interceptor: Option<Interceptor> = None;
```

Comment:
No need to `mut` interceptor. You can write like below:

```Rust
let interceptor = if let xxx {
    // ...
} else {
    //...
};
```

Reply:
OK